packages/vite/src/node/server/middlewares/hostCheck.ts

```diff
@@ -91,6 +91,11 @@ export function isHostAllowedWithoutCache(
   if (hostname === 'localhost' || hostname.endsWith('.localhost')) {
     return true
   }
+  // allow test and .test by default as they will never be registered
+  // https://datatracker.ietf.org/doc/html/rfc6761#section-6.2
+  if (hostname === 'test' || hostname.endsWith('.test')) {
+    return true
+  }
```

Comment:
For the `test` domain name / hostname, the document mentions:

> 2. Application software SHOULD NOT recognize test names as special,
> and SHOULD use test names as they would other domain names.

Should we not handle the test hostname in that case?

For TLDs I think it's fine since it doesn't mention anything like that https://datatracker.ietf.org/doc/rfc2606/

Reply:
I think that sentence is meant to include both `test` and `*.test`s.

> The domain "test.", and any names falling within ".test.", are special in the following ways:

(FYI the last dot of `test.`/`.test.` is to [clarify that it's the last item](https://en.wikipedia.org/wiki/Fully_qualified_domain_name))

Reply:
Maybe we should not add `.test` anyway as trusting the DNS server does not necessarily mean you can trust the IPs listed in that DNS server.

Reply:
I see. Then I guess technically it discourages treating `test` TLDs differently, so I think I'm somewhat leaning on not adding `.test` as a special case 🤔

Reply:
Let's say there's a internal DNS server and you and your team all uses that DNS server. Then, a member of your team added `foo.test` that points to some IP. You know that the DNS server would not fool you, but don't know whether the IP `foo.test` is pointing is trustable. In most cases, you would never add a DNS record that points to an IP that you don't hold, but you can point to an IP you don't have control of.
You may say that you should never put an IP you don't control, but it's not enforced by something, and there maybe a user that is using `.test` like that.

Reply:
> Let's say there's an internal DNS server and you and your team all uses that DNS server. Then, a member of your team added `foo.test` that points to some IP. (...)

Thanks for explanation. I just thought there would be system level safeguards that would prevent any modern system from ever resolving the `.test` domain via DNS. But I am probably asking for too much here and we cannot count on such (potential) system level implementations, and anyway that would be against the [rfc6761](https://datatracker.ietf.org/doc/html/rfc6761#section-6.2) `Application software SHOULD NOT recognize test names as special` principle.